src/Symfony/Bundle/FrameworkBundle/Command/SecretsListCommand.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Bundle\FrameworkBundle\Exception\EncryptionKeyNotFoundException;
+use Symfony\Bundle\FrameworkBundle\Secret\SecretStorageInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * @author Tobias Schultze <http://tobion.de>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+final class SecretsListCommand extends Command
+{
+    protected static $defaultName = 'secrets:list';
+
+    private $secretStorage;
+
+    public function __construct(SecretStorageInterface $secretStorage)
+    {
+        $this->secretStorage = $secretStorage;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('Lists all secrets.')
+            ->addOption('reveal', 'r', InputOption::VALUE_NONE, 'Display plain text value along side names');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $reveal = $input->getOption('reveal');
+        $io = new SymfonyStyle($input, $output);
+
+        try {
+            $secrets = $this->secretStorage->listSecrets($reveal);
+        } catch (EncryptionKeyNotFoundException $e) {
+            throw new \LogicException(sprintf('Unable to decrypt secrets, the encryption key "%s" is missing.', $e->getKeyLocation()));
+        }
+
+        if ($reveal) {
+            $rows = [];
+            foreach ($secrets as $name => $value) {
+                $rows[] = [$name, $value];
+            }
+            $io->table(['name', 'plain text secret'], $rows);
+
+            return;
```

Comment:
readability is weird
what about an if else ?

or simply one loop with a check if $reveal then you use the $value instead of $_

Reply:
I'm not sure it will be more readable. Could you provide a snippet?

Reply:
@jderusse something like

```php
$rows = [];
foreach ($secrets as $name => $value) {
    $rows[] = [$name, $reveal ? $value : '***'];
}

$io->comment(sprintf('To reveal the values of the secrets, use <info>php %s %s --reveal</info>', $_SERVER['PHP_SELF'], $this->getName()));
$io->table(['name', 'secret'], $rows);
```

note we can also write conditionionaly the $io->comment, but keeping it always is ok imo

Reply:
I prefer right now